### ISSUE TITLE
[Bug] Failure to convert EfficientDet-D0 TF model to IR with input_size (NHWC) containing N > 1

### ISSUE THREAD

**Issue by sumeethn** [openvinotoolkit/openvino#11962]
##### System information (version)
- OpenVINO => 2022.1
- Operating System / Platform => Ubuntu 20.4 (openvino/ubuntu20_dev:latest)
- Problem classification: Model Conversion
- Framework: TensorFlow
- Model name: EfficentDet-D0

##### Detailed description
Need to convert EfficientDet-D0 TF model to IR, specifically, to run IR model with a batch size of 2. Model optimizer fails with the error 

openvino@c7568e075e16:/omz_model/public/efficientdet-d0-tf/efficientdet-d0_saved_model$ mo --input_model efficientdet-d0_frozen.pb --transformations_config /usr/local/lib/python3.8/dist-packages/openvino/tools/mo/front/tf/automl_efficientdet.json --input_shape [2,512,512,3]
Model Optimizer arguments:
Common parameters:
	- Path to the Input Model: 	/omz_model/public/efficientdet-d0-tf/efficientdet-d0_saved_model/efficientdet-d0_frozen.pb
	- Path for generated IR: 	/omz_model/public/efficientdet-d0-tf/efficientdet-d0_saved_model/.
	- IR output name: 	efficientdet-d0_frozen
	- Log level: 	ERROR
	- Batch: 	Not specified, inherited from the model
	- Input layers: 	Not specified, inherited from the model
	- Output layers: 	Not specified, inherited from the model
	- Input shapes: 	[2,512,512,3]
	- Source layout: 	Not specified
	- Target layout: 	Not specified
	- Layout: 	Not specified
	- Mean values: 	Not specified
	- Scale values: 	Not specified
	- Scale factor: 	Not specified
	- Precision of IR: 	FP32
	- Enable fusing: 	True
	- User transformations: 	Not specified
	- Reverse input channels: 	False
	- Enable IR generation for fixed input shape: 	False
	- Use the transformations config file: 	/usr/local/lib/python3.8/dist-packages/openvino/tools/mo/front/tf/automl_efficientdet.json
Advanced parameters:
	- Force the usage of legacy Frontend of Model Optimizer for model conversion into IR: 	False
	- Force the usage of new Frontend of Model Optimizer for model conversion into IR: 	False
TensorFlow specific parameters:
	- Input model in text protobuf format: 	False
	- Path to model dump for TensorBoard: 	None
	- List of shared libraries with TensorFlow custom layers implementation: 	None
	- Update the configuration file with input/output node names: 	None
	- Use configuration file used to generate the model with Object Detection API: 	None
	- Use the config file: 	None
OpenVINO runtime found in: 	/opt/intel/openvino/python/python3.8/openvino
OpenVINO runtime version: 	2022.1.0-7019-cdb9bec7210-releases/2022/1
Model Optimizer version: 	2022.1.0-7019-cdb9bec7210-releases/2022/1
[ ERROR ]  Cannot infer shapes or values for node "detections".
[ ERROR ]  Locations and prior boxes shapes mismatch: "[     1 392832]" vs "[     1      2 196416]" for node "detections"
[ ERROR ]
[ ERROR ]  It can happen due to bug in custom shape infer function <function DetectionOutput.infer at 0x7f4e858935e0>.
[ ERROR ]  Or because the node inputs have incorrect values/shapes.
[ ERROR ]  Or because input shapes are incorrect (embedded to the model or passed via --input_shape).
[ ERROR ]  Run Model Optimizer with --log_level=DEBUG for more information.
[ ERROR ]  Exception occurred during running replacer "REPLACEMENT_ID" (<class 'openvino.tools.mo.middle.PartialInfer.PartialInfer'>): Stopped shape/value propagation at "detections" node.
 For more information please refer to Model Optimizer FAQ, question #38. (https://docs.openvino.ai/latest/openvino_docs_MO_DG_prepare_model_Model_Optimizer_FAQ.html?question=38#question-38)

##### Steps to reproduce

1. Downloaded EfficientDet-D0 model using omz_downloader2. 
2. Converted the model to efficientdet-d0_frozen.pb 
3. Used mo with options to convert EfficientDet-D0 to IR:4. The line below is executed in openvino/ubuntu20_dev:latest container
`mo --input_model efficientdet-d0_frozen.pb --transformations_config /usr/local/lib/python3.8/dist-packages/openvino/tools/mo/front/tf/automl_efficientdet.json --input_shape [2,512,512,3]` 
4. The efficientdet-d0_frozen.pb is attached.

##### Issue submission checklist

 - [ X] I report the issue, it's not a question
   
 - [ X] I checked the problem with documentation, FAQ, open issues, Stack Overflow, etc and have not found solution
   
 - [ ] There is reproducer code and related data files: images, videos, models, etc.
   
[efficientdet-d0_frozen.pb.zip](https://github.com/openvinotoolkit/openvino/files/8974191/efficientdet-d0_frozen.pb.zip)

**Comment by Iffa-Meah** [comment]
@sumeethn your issue is related to the input shape that you parse. The **N** value of NCHW should not be other that 1

Your command: 
mo --input_model efficientdet-d0_frozen.pb --transformations_config /usr/local/lib/python3.8/dist-packages/openvino/tools/mo/front/tf/automl_efficientdet.json --input_shape [2,512,512,3]

Try to change to:
mo --input_model efficientdet-d0_frozen.pb --transformations_config /usr/local/lib/python3.8/dist-packages/openvino/tools/mo/front/tf/automl_efficientdet.json --input_shape [1,512,512,3]

This is an example of conversion using official omz_converter. Check out the input shape part (same case fro FP16 or FP31)
![InputShape](https://user-images.githubusercontent.com/79643476/175894025-f60e8a1c-7c05-4046-babd-567cceb7603e.png)

**Comment by brmarkus** [comment]
@sumeethn can you maybe rephrase your question?
You have a model which currently is not supporting batch-mode as "N=1" - and now you want to make the model supporting batches for N>1? So you want to convert the model to IR format supporting batch-mode, a fixed batch-size or dynamic batch size?

**Comment by Iffa-Meah** [comment]
@sumeethn, in that case, you can convert your model into a dynamic IR.
Example for dynamic IR conversion command: mo -m yourModelPath --input_shape [-1,3,256,100] --data_type FP32
(Note that the -1 or ? symbol would indicate the dynamic shape).
Parse in your own input shape & desired data type.

Example of inferencing command: 

benchmark_app -m mbnetv3_embeddings.xml -data_shape [1,3,256,100]
benchmark_app -m mbnetv3_embeddings.xml -data_shape [2,3,256,100]
benchmark_app -m mbnetv3_embeddings.xml -data_shape [3,3,256,100]

OR

benchmark_app -m mbnetv3_embeddings.xml -shape [1,3,256,100]
benchmark_app -m mbnetv3_embeddings.xml -shape [2,3,256,100]
benchmark_app -m mbnetv3_embeddings.xml -shape [3,3,256,100]

**Comment by sumeethn** [comment]
Thank you! I was able to convert the model into dynamic shape and load the model into OpenVINO model server. I would like to run the benchmark on OVMS. I will close this issue.
